sdk/iot/hub/samples/telemetry/src/iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,278 @@
+
+// AZ_IOT_DEVICE_X509_CERT_PEM_FILE is the path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+#define DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+// AZ_IOT_DEVICE_X509_TRUST_PEM_FILE is the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+#define NUMBER_OF_MESSAGES 5
+
+static char device_id[64];
+static char iot_hub_fqdn[128];
+static char x509_cert_pem_file[512];
+static char x509_trust_pem_file[256];
+char telemetry_topic[128];
+
+static char mqtt_endpoint[128];
+static az_span mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("ssl://");
+static az_span mqtt_url_suffix = AZ_SPAN_LITERAL_FROM_STR(":8883");
+
+static const char* telemetry_message_payloads[NUMBER_OF_MESSAGES] = {
+  "Message One", "Message Two", "Message Three", "Message Four", "Message Five",
+};
+
+static az_iot_hub_client client;
+static MQTTClient mqtt_client;
+
+static void sleep_seconds(uint32_t seconds)
+{
+#ifdef _WIN32
+  Sleep((DWORD)seconds * 1000);
+#else
+  sleep(seconds);
+#endif
+}
+
+static az_result read_configuration_entry(
+    const char* name,
+    const char* env_name,
+    char* default_value,
+    bool hide_value,
+    az_span buffer,
+    az_span* out_value)
+{
+  printf("%s = ", name);
+  char* env = getenv(env_name);
+
+  if (env != NULL)
+  {
+    printf("%s\n", hide_value ? "***" : env);
+    az_span env_span = az_span_from_str(env);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(env_span));
+    az_span_copy(buffer, env_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(env_span));
+  }
+  else if (default_value != NULL)
+  {
+    printf("%s\n", default_value);
+    az_span default_span = az_span_from_str(default_value);
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(buffer, az_span_size(default_span));
+    az_span_copy(buffer, default_span);
+    *out_value = az_span_slice(buffer, 0, az_span_size(default_span));
+  }
+  else
+  {
+    printf("(missing) Please set the %s environment variable.\n", env_name);
+    return AZ_ERROR_ARG;
+  }
+
+  return AZ_OK;
+}
+
+static az_result create_mqtt_endpoint(char* destination, int32_t size, az_span iot_hub)
+{
+  int32_t iot_hub_length = (int32_t)strlen(iot_hub_fqdn);
+  int32_t required_size = az_span_size(mqtt_url_prefix) + iot_hub_length + az_span_size(mqtt_url_suffix);
+  if (required_size > size)
+  {
+    return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
+  }
+  az_span destination_span = az_span_init((uint8_t*)destination, size);
+  az_span remainder = az_span_copy(destination_span, mqtt_url_prefix);
+  remainder = az_span_copy(remainder, az_span_slice(iot_hub, 0, iot_hub_length));
+  az_span_copy(remainder, mqtt_url_suffix);
+
+  return AZ_OK;
+}
+
+static az_result read_configuration_and_init_client()
```

Comment:
>read_configuration_and_init_client [](start = 17, length = 34)

init_clien_using_configuration?

Reply:
its the same elsewhere so for now I'm inclined to keep as is.

Reply:
I think this is how provisioning has it named. If we change, please update all samples.